### ISSUE TITLE
[UI] Browser refreshing and the unholy trailing slash

### ISSUE THREAD

**Issue by uepoch** [hashicorp/vault#4660]
**Describe the bug**
When your browser(Chrome/FF) open an URL, it does trim the last '/' which leave the list page in a weird state, the filter being likely filled by the last part of the url, it prepend the last directory to the items in the list, and thus make the link to them invalid
Not sure how it's done in vault, but browsers seem to accept that js modify the uri to something that ends with / 
**To Reproduce**
```
vault server -dev -dev-root-token-id foo &
export VAULT_ADDR=http://localhost:8200
export VAULT_TOKEN=foo
sleep 3
vault kv put secret/a/b/c e=f
echo '{"path":{"/*":{"capabilities":["read","update","create","list","delete"]}}}' > /tmp/policy.json
vault policy write test /tmp/policy.json
vault token create -policy test
fg
```

Then login using the token given and go to any dir.
Press refresh button
or directly load an url like http://localhost:8200/ui/vault/secrets/secret/list/a/b/

Try to click on an item -> Links to secret/a/bc/

**Expected behavior**
It returns the correct entries

**Environment:**
* Vault Server Version (retrieve with `vault status`): upstream ( testing with 0.10 too ) 
* Vault CLI Version (retrieve with `vault version`): upstream ( tested with 0.10 too )
* Server Operating System/Architecture: Linux/Osx

Vault server configuration file(s): dev

**Comment by jefferai** [comment]
Possibly duplicate of https://github.com/hashicorp/vault/issues/4581 ? @meirish ?

**Comment by meirish** [comment]
Unrelated (sadly) - though both aren't caught with the ember proxy server running. Almost have a PR though!

**Comment by briankassouf** [comment]
Closed via #4668
